app/services/geo_resources/discovery/document_builder/date_builder.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+module GeoResources
+  module Discovery
+    class DocumentBuilder
+      class DateBuilder
+        attr_reader :resource_decorator
+
+        def initialize(resource_decorator)
+          @resource_decorator = resource_decorator
+        end
+
+        # Builds date fields such as layer year and modified date.
+        # @param [AbstractDocument] discovery document
+        def build(document)
+          document.layer_year = layer_year
+          document.layer_modified = layer_modified
+          document.issued = issued
+        end
+
+        private
+
+          # Returns the date the layer was issued.
+          # @return [String] date in XMLSchema format.
+          def issued
+            datetime = resource_decorator.issued.first
+            datetime = DateTime.parse(datetime).utc
+            datetime.utc.xmlschema
+          rescue
+            ''
```

Comment:
Should a warning be logged for cases where the datestamp cannot be normalized and formatted?

Reply:
I'm not terribly worried about a missing issued property because of a malformed date. We don't even display the field in Pulmap. Probably good to add it to our docs, if we can, but no big deal if we can't.